src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2AuthenticationContext.cs

```diff
@@ -73,8 +73,14 @@ public Saml2AuthenticationContext(Uri classReference)
         /// <param name="declarationReference">The declaration reference of the authentication context.</param>
         public Saml2AuthenticationContext(Uri classReference, Uri declarationReference)
         {
-            ClassReference = classReference;
-            DeclarationReference = declarationReference;
+            // Class and declaration reference are optional, so don't try and set the respective properties if
```

Comment:
I would recommend a slightly different pattern.
We have three ctors.
Check the parameters as we did before, but call a private method to initialize where we know the parameters were checked.
Otherwise we are not helping devs by helping them catch programming errors.

Here is an example here where we know in the private method we can assume parameters were checked in the public  methods.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/2cb5b3e600b0e9f978f1d468f096e17fd5c1bc5f/src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs#L454

Reply:
I took a slightly different approach, but hopefully captured the spirit of what your feedback was addressing.

Reply:
Yep, looks good.